actix-http/src/config.rs

```diff
@@ -211,7 +212,7 @@ impl Date {
     }
     fn update(&mut self) {
         self.pos = 0;
-        write!(self, "{}", time::at_utc(time::get_time()).rfc822()).unwrap();
+        write!(self, "{}", PrimitiveDateTime::now().format("%a, %d %b %Y %H:%M:%S GMT")).unwrap();
```

Comment:
time must be in UtcOffset::UTC offset.

Reply:
For this, I took `rfc822`'s format pattern of `%a, %d %b %Y %T GMT` (i.e. `tm_utcoff` = 0) and used that. Being that this section of code deals with updating the value used in the Date header, I compared it to what's shown from inspecting a Date header in the latest Firefox to ensure that it made sense.

Do you mean that it should always end in UTC (which goes against what I saw in-browser, where it was always GMT, although there should be no difference), or do you mean that it should be a UTC offset ('+0100' for example, using format pattern `%z` (i.e. `rfc822z` from time 0.1))?

Reply:
Time itself must be in utc, otherwise you create time in local time zone and then mark it as gmt

Reply:
time 0.2 has no concept of local time or system time. PrimitiveDateTime always is UTC, and PrimitiveDateTime::now() will always return a UTC-based date and time. This differs from time 0.1's `time::now()` and `time::at_utc(time::get_time())` functions, as `time::now()` would have returned the local time.

Reply:
Oh, this is fucked up. Nice.

Reply:
[This closed `time` issue explains the change to always-UTC.](https://github.com/time-rs/time/issues/190)

Reply:
I guess I should also clarify and mention that there *is* a way to get the system's time, but to do that, you'd do:

```Rust
PrimitiveDateTime::from(std::time::SystemTime::now())
```